### DISCUSSION TITLE
Docker: How to run hyundai_kia_connect_monitor with cron

### DISCUSSION THREAD

**Discussion by Andre1Becker** [ZuinigeRijder/hyundai_kia_connect_monitor#56] [Show and tell]
Hi All,
this is just for documenting on how i did get this nice little script running in a docker container.
1. Download [hyundai_kia_connect_api](https://github.com/Hyundai-Kia-Connect/hyundai_kia_connect_api/releases)
2. Download [latest hyundai_kia_connect_monitor release](https://github.com/ZuinigeRijder/hyundai_kia_connect_monitor/releases)
3. Create a folder-structure like this:
     MyDockerFolder
               -> data
               -> app
4. Extract `hyundai_kia_connect_monitor-Rx.xx.x.zip` to `MyDockerFolder/app`
5. Extract `hyundai_kia_connect_api-x.xx.x.zip` to `MyDockerFolder/app/hyundai_kia_connect_api`
6. Create a file called `crontab.txt` in `MyDockerFolder/`
7. Paste this in `crontab.txt`:
```
HOME=/data
PATH=/usr/local/bin:/usr/bin:/bin:/usr/local/games:/usr/games
*/15 07-20 * * * python3 /opt/hyundai_monitor/monitor.py >> /var/log/monitor.log #Hyundai Monitoring
0 20-07 * * * python3 /opt/hyundai_monitor/monitor.py >> /var/log/monitor.log #Hyundai Monitoring
```
8. Create a file named `entry.sh` in `MyDockerFolder/app/`
9. Paste this in `entry.sh`:
```
#!/bin/sh
# start cron
/usr/sbin/cron -f -l 8
```
10. Create a `dockerfile` in `MyDockerFolder/`
11. Paste this in `dockerfile`
```
FROM python:3.11
ADD app /opt/hyundai_monitor/
ADD data /data/
ADD crontab.txt /crontab.txt
RUN apt-get update
RUN apt-get install -y cron bash curl
RUN pip install beautifulsoup4 python_dateutil pytz requests gspread geopy cloudscraper paho-mqtt tzdata certifi
VOLUME /data
RUN chmod +x /opt/hyundai_monitor/entry.sh
RUN /usr/bin/crontab /crontab.txt
CMD ["./opt/hyundai_monitor/entry.sh"]
```
12. Move `monitor.cfg` from `MyDockerFolder/app` to `MyDockerFolder/data`

The next step is not needed anymore since [hyundai_kia_connect_monitor R3.18.0](https://github.com/ZuinigeRijder/hyundai_kia_connect_monitor/releases/tag/R3.18.0)
~13. Modify `monitor.py` to set the path to `monitor.cfg`~
14. Modify `monitor.cfg` for your car, region, username and password

Finally your folder-structure should like a bit like this:
![image](https://github.com/ZuinigeRijder/hyundai_kia_connect_monitor/assets/25841903/a1c90091-a535-4cce-a129-5465b73c3cdd)

15. Thats it. Now use  `docker build -t hyundai/hyundai_monitor .` to generate your docker container. 
16. Optional: on Unraid you have to tag und push the container: 
docker image tag hyundai/hyundai_monitor localhost:5000/hyundai_monitor
docker push localhost:5000/hyundai_monitor
Happy Tracking!

Also, you could mount `data` to a folder outside of docker to access all csv's as you like.

Any suggestions are welcome. I'm no expert in any of this, just figuring stuff out as i go. Thanks to all people from many guides and forum posts that i found to get this running ;-)

And a big shout out to the dev('s) from [Hyundai-Kia-Connect](https://github.com/Hyundai-Kia-Connect) and @ZuinigeRijder for creating this script in the first place!

**Comment by ZuinigeRijder** [comment]
Probably I need to fix the reading of .cfg files, to first look into current directory and then look into the python script directory. monitor.cfg is also read in the other scripts and also summary.cfg and translations.csv. Then you do not have to modify the python scripts.

I will make a release of this change in the near future.

**Reply by Andre1Becker** [reply]
That cloud make this process a bit easier ;-) Thanks!
BTW - I just did that so i could modify the `monitor.cfg` if i make a mistake - or if someone else using this little "tutorial" to not bake in any credentials in this file.

**Comment by ZuinigeRijder** [comment]
@Andre1Becker I made a release for searching the .cfg files first in the current directory, then in the python script directory: [R3.18.0](https://github.com/ZuinigeRijder/hyundai_kia_connect_monitor/releases/tag/R3.18.0)
